kivy_ios/toolchain.py

```diff
@@ -209,8 +209,17 @@ def get_env(self):
                  'include_file_mtime,include_file_ctime,file_stat_matches'))
 
         if not self._ccsh:
-            self._ccsh = tempfile.NamedTemporaryFile()
-            self._cxxsh = tempfile.NamedTemporaryFile()
+            def noicctempfile():
+                '''
+                reported issue where C Python has issues with 'icc' in the compiler path
+                '''
+                x = tempfile.NamedTemporaryFile()
+                while 'icc' in x.name:
+                    x = tempfile.NamedTemporaryFile()
+                return x
```

Comment:
I'm wondering if the walrus operator could help make it more concise (untested):
```python
while 'icc' in (x := tempfile.NamedTemporaryFile()).name:
    pass
return x
```
Feel free to keep your version if you think it's more readable